### PR TITLE
BUG: Fixes default progress callback raises user warning in send() function

### PR DESCRIPTION
**Pull request by marcus-wirtz-snkeos**
Since the return argument of the `defaultProgressCallback()` is `None`, the `send()` function of the `DICOMSender` raises a UserWarning:
```python
if not self.progressCallback(f"Sending {file} to {self.destinationUrl.toString()} using {self.protocol}"):
    raise UserWarning("Sending was cancelled, upload is incomplete.")
```

This minimal PR prevents the user warning by returning True instead.

### CODE REVIEW

File: Modules/Scripted/DICOMLib/DICOMProcesses.py [hunk 1]
```diff
@@ -418,8 +418,10 @@ def __init__(self, files, address, protocol=None, progressCallback=None, aeTitle
     def __del__(self):
         super().__del__()
 
-    def defaultProgressCallback(self, s):
+    @staticmethod
```
Comment (lassoan) [inline Modules/Scripted/DICOMLib/DICOMProcesses.py:421]: Callback is usually not static method, as it uses other methods of the object to display the progress information, determine if the operation should be cancelled, etc. What was the reason for making it staticmethod?

**Review by lassoan — COMMENTED** [review]
Thank you, good catch, returning True is the right thing to do.

You need to fix the commit message (it must start with a capital letter after `BUG: `) and please also see the comment on the necessity of changing the callback to a static method.

File: Modules/Scripted/DICOMLib/DICOMProcesses.py (hunk 1 above)
Comment (marcus-wirtz-snkeos) [reply Modules/Scripted/DICOMLib/DICOMProcesses.py:421]: True, this particular function was not using `self`, that was the only reason. But I agree that it should inherit the same architecture -> I'll update this.

**Comment by marcus-wirtz-snkeos** [comment]
@lassoan sorry, I messed up the commit history by merging main, here is the updated PR:
https://github.com/Slicer/Slicer/pull/6822

**Pull request closed without merging**